chore: Add docs on local prompt template (#66)
SAP/ai-sdk: docs-js/orchestration/chat-completion.mdx

});
```

> ### Remote Prompt Registry

Author: I mentioned it so the difference is clear. But i see your point.

> ### Remote Prompt Registry

Author: ```suggestion
### Prompt Registry
```

> You can define template content within the templating configuration, or refer to a template defined in the [Prompt Registry](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/prompt-registry).

Reviewer: ```suggestion
You can refer to a template defined in the [Prompt Registry](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/prompt-registry) instead of defining within the `templating` configuration.
```

```ts
const orchestrationClient = new OrchestrationClient({
  llm,
  templating: {
    template_ref: {
      name: 'my-get-capital-template',
      scenario: 'my-scenario',
      version: '0.0.1'
    }
  }
});

const response = orchestrationClient.chatCompletion({
  inputParams: { country: 'France' }
});
```

> A prompt template can be referenced either by ID, or by a combination of name, scenario and version.

Author: The id is a long guid, so honestly I am not even sure who would ever remember/use it.

For details on storing a template in the Prompt Registry, refer to [this guide](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-prompt-template-imperative).

### Local Prompt Template

> You can define the template content by passing the contents of a YAML file as a string in the `templating` configuration.

Reviewer: ```suggestion
You can define the template in a YAML file and pass its content as a string to the `templating` property.
```

This is useful for testing a prompt template locally before storing it in the [Prompt Registry](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/prompt-registry).

```ts
import { readFileSync } from 'fs';
import { OrchestrationClient } from '@sap-ai-sdk/orchestration';

// Read the YAML file containing the prompt template

> const yamlTemplate = readFileSync('./path/to/prompt-template.yaml', 'utf-8');

Reviewer 2: Replace with `readFile`?

const orchestrationClient = new OrchestrationClient({
  llm,
  templating: yamlTemplate
});

const response = orchestrationClient.chatCompletion({
  inputParams: { country: 'France' }
});
```

> The YAML string is parsed and validated against the expected schema.

Reviewer 2: the expected schema of a prompt template? can we link it here?

If the YAML is invalid or does not conform to the schema, an error will be thrown.

> This [YAML file](https://github.com/SAP/ai-sdk-js/blob/main/test-util/data/orchestration/orchestration-chat-completion-yaml-template.yaml) illustrates the format in which a prompt template can be defined for use in the templating configuration.

Author: ```suggestion
This [section](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-prompt-template-declarative#procedure) shows the format in which a prompt template can be defined.
```

> This [YAML file](https://github.com/SAP/ai-sdk-js/blob/main/test-util/data/orchestration/orchestration-chat-completion-yaml-template.yaml) illustrates the format in which a prompt template can be defined for use in the templating configuration.

Reviewer: [q] I wonder if we should link the schema here so that user knows how to construct such YAML files? `PromptTemplatePostRequestSchema`

Author: Good idea, I think we do link to some official help.sap docs here. I'll check and add if its missing
